crates/proto/src/rr/rdata/mx.rs

```diff
@@ -84,12 +84,9 @@ impl MX {
 
 impl BinEncodable for MX {
     fn emit(&self, encoder: &mut BinEncoder<'_>) -> ProtoResult<()> {
-        let is_canonical_names = encoder.is_canonical_names();
         encoder.emit_u16(self.preference())?;
 
-        // to_lowercase for rfc4034 and rfc6840
-        self.exchange()
-            .emit_with_lowercase(encoder, is_canonical_names)?;
+        self.exchange().emit(encoder)?;
```

Comment:
I think this commit has too much going on. It looks like hunks like this could be extracted into a separate commit?